packages/storybook/src/builders/build-storybook/build-storybook.impl.ts

```diff
@@ -87,8 +87,7 @@ async function storybookOptionMapper(
     configDir: storybookConfig,
     ...frameworkOptions,
     frameworkPresets: [...(frameworkOptions.frameworkPresets || [])],
-    watch: false,
-    docsMode: builderOptions.docsMode
```

Comment:
This isn't required because all builder options are spread into this object a few lines above. I removed it to keep both builders consistent.